deeplearning4j-scaleout/spark/dl4j-spark/src/main/spark-1/org/deeplearning4j/spark/util/BaseDoubleFlatMapFunctionAdaptee.java

```diff
@@ -0,0 +1,25 @@
+package org.deeplearning4j.spark.util;
+
+import org.apache.spark.api.java.function.DoubleFlatMapFunction;
+
+import java.util.Iterator;
+
+/**
+ * DoubleFlatMapFunction adapter to hide incompatibilities between Spark 1.x and Spark 2.x
+ *
+ * This class should be used instead of direct referral to DoubleFlatMapFunction
+ *
+ */
+public class BaseDoubleFlatMapFunctionAdaptee<T> implements DoubleFlatMapFunction<T> {
```

Comment:
It's deeply unsatisfying how un-generic this is, simply for a return type's member. Do you feel like factoring the `BaseUFlatMapFunctionAdapter<U, T> Implements UFlatMapFunction<U, T>` ?